src/librustc_codegen_llvm/common.rs

```diff
@@ -456,3 +456,22 @@ pub fn ty_fn_sig<'a, 'tcx>(cx: &CodegenCx<'a, 'tcx>,
         _ => bug!("unexpected type {:?} to ty_fn_sig", ty)
     }
 }
+
+pub fn ty_fn_sig_vtable<'a, 'tcx>(
```

Comment:
I would expect this to be on `Instance` itself. Same for `ty_fn_sig` above, I guess. Feel free to leave `FIXME` comments here instead.

Reply:
Should I introduce `InstanceExt` in codegen_llvm, or just move them into librustc?

Reply:
This feels like it would fit in librustc.

Reply:
Moved `ty_fn_sig(_vtable)` under `Instance` as `Instance::fn_sig`. Now `is_vtable_shim` is seen in only a few places.